rdt/transformers/categorical.py

```diff
@@ -472,7 +478,10 @@ def _transform(self, data):
             size=is_null.sum()
         )
 
-        return mapped.astype('int64')
+        if self.add_noise:
+            mapped = np.random.uniform(mapped, mapped + 1)
```

Comment:
since we already use the `pandas.Series.map` function, can we do this there and just pass a function that either uses the dictionary value if `add_noise` is false or uniformly samples from the dictionary value otherwise?

Reply:
The issue with this change are lines 476-479:
```python3
mapped[is_null] = np.random.randint(
    len(self.categories_to_values),
    size=is_null.sum()
)
```

If I move the `self.add_noise`, these lines will have to be rewritten to something like:
```python3
if self.add_noise:
    mapped[is_null] = np.random.uniform(
        0,
        len(self.categories_to_values),
        size=is_null.sum()
    )
else:
    mapped[is_null] = np.random.randint(
        len(self.categories_to_values),
        size=is_null.sum()
    )
```
I think the speed gain is negligible compared to the loss in code readability.